src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -237,6 +241,20 @@ public IEnumerable<char> PotentialCommitCharacters
                 return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
+            // Specifically for snippets, we use reflection to try and invoke the LanguageServerSnippetExpander's
+            // TryExpand method and determine if it succeeded or not.
+            if (SnippetCompletionItem.IsSnippet(roslynItem))
+            {
+                var lspSnippetText = change.Properties[SnippetCompletionItem.LSPSnippetKey];
+
+                if (!_roslynLSPSnippetExpander.TryExpand(change.TextChange.Span, lspSnippetText, _textView, triggerSnapshot))
+                {
+                    return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
+                }
+
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
```

Comment:
i don't see any difference with teh true/false branches of hte 'if' statement.  if there is no difference, remove hte if check (and consider making TryExpand not return a boolean as you don't need it for anything).